sql/sql_acl.cc

```diff
@@ -3614,18 +3614,18 @@ static void init_check_host(void)
                                sizeof(struct acl_host_and_ip),
                                acl_users.elements, 1, MYF(0));
   (void) my_hash_init(key_memory_acl_mem, &acl_check_hosts,system_charset_info,
-                      acl_users.elements, 0, 0,
+                      (ulong)acl_users.elements, 0, 0,
```

Comment:
Looks like this should be a follow-up fix. It will be hard to remember to remove this downcast later when doing my_hash_init change, so can you also update *just* `my_hash_init2` function prototype to take in a size_t for that particular parameter?

Then, when we refactor HASH to use size_t as well, we can fix it and not forget about this cast.